### DISCUSSION TITLE
auto fix shapes and you are a god

### DISCUSSION THREAD

**Discussion by Wladefant** [personalizedrefrigerator/js-draw#49] [Ideas & Feature Requests]
One of the only things I am missing from samsung notes are auto fix shapes, everything with them is perfect, I suppose firstly a model needs to be trained and that would eat ressources, but still it is a very important feature. 
![SmartSelect_20231027_150957_Samsung Notes](https://github.com/personalizedrefrigerator/js-draw/assets/79611093/90474438-b40a-4181-b840-676e3a64dfa6)

https://github.com/personalizedrefrigerator/js-draw/assets/79611093/8637628a-2f14-4d1e-960a-034b93338276

**Comment by personalizedrefrigerator** [comment]
I think [Xournal++ is able to do it without training a model](https://github.com/xournalpp/xournalpp/tree/master/src/core/control/shaperecognizer), so maybe js-draw could, too.

**Comment by personalizedrefrigerator** [comment]
Implemented for lines and (some) rectangles in https://github.com/personalizedrefrigerator/js-draw/commit/e627d1f754c59c9454a76274836648c2fb82f05e

**Reply by Wladefant** [reply]
Lines is the most important for sure

**Reply by personalizedrefrigerator** [reply]
This is the current UI for enabling/disabling autocorrect: 

![screenshot: two buttons at the bottom of the pen dialog: autocorrect and stabilization](https://github.com/personalizedrefrigerator/js-draw/assets/46334387/e4448ca5-8460-4c02-9216-25bd61afc7fd)

I think the spacing/sizing of the buttons still needs to be adjusted, but after that, it should be ready for release.

**Reply by personalizedrefrigerator** [reply]
https://github.com/personalizedrefrigerator/joplin/releases/tag/prerelease-3
